From 577f38e092f9847f1b5535fc9f8b736cc441e8ea Mon Sep 17 00:00:00 2001
From: Rick Bernotas <rbernota@yahoo-inc.com>
Date: Wed, 29 Apr 2015 13:26:12 -0500
Subject: [PATCH 1163/1173] HUE-2716 [pig] Scripts fail on hcat auth with
 org.apache.hive.hcatalog.pig.HCatLoader()

Fix for HUE-2716, updating use_hcatalog() in pig's models.py to set hcat credentials
for org.apache.hive.hcatalog.pig.HCatLoader and HCatStorer, in addition to the
soon-to-be-deprecated api's org.apache.hcatalog.pig.HCatLoader and HCatStorer.
---
 apps/pig/src/pig/models.py |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/apps/pig/src/pig/models.py b/apps/pig/src/pig/models.py
index 026ed14..20f9e90 100644
--- a/apps/pig/src/pig/models.py
+++ b/apps/pig/src/pig/models.py
@@ -81,8 +81,8 @@ class PigScript(Document):
   @property
   def use_hcatalog(self):
     script = self.dict['script']
-    return 'org.apache.hcatalog.pig.HCatStorer' in script or 'org.apache.hcatalog.pig.HCatLoader' in script
-
+    return ('org.apache.hcatalog.pig.HCatStorer' in script or 'org.apache.hcatalog.pig.HCatLoader' in script) or \ # Deprecated
+    ('org.apache.hive.hcatalog.pig.HCatLoader' in script or 'org.apache.hive.hcatalog.pig.HCatStorer' in script)
 
 def create_or_update_script(id, name, script, user, parameters, resources, hadoopProperties, is_design=True):
   try:
-- 
1.7.9.5

